src/Tests/Microsoft.Diagnostics.Monitoring.Profiler.UnitTests/Baselines/ExceptionThrowCrash.txt

```diff
@@ -0,0 +1,13 @@
+﻿[profiler]dbug: Exception thrown: Microsoft.Diagnostics.Monitoring.Profiler.UnitTestApp.dll!Microsoft.Diagnostics.Monitoring.Profiler.UnitTestApp.Scenarios.ExceptionThrowCrashScenario.Execute
+[profiler]dbug: Exception handled: System.CommandLine.dll!System.CommandLine.Invocation.AnonymousCommandHandler+<InvokeAsync>d__6.MoveNext
+[profiler]dbug: Exception thrown: System.Private.CoreLib.dll!System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw
+[profiler]dbug: Exception handled: System.CommandLine.dll!System.CommandLine.Parsing.ParseResultExtensions+<InvokeAsync>d__0.MoveNext
+[profiler]dbug: Exception thrown: System.Private.CoreLib.dll!System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw
+[profiler]dbug: Exception handled: System.CommandLine.dll!System.CommandLine.Parsing.ParserExtensions+<InvokeAsync>d__3.MoveNext
+[profiler]dbug: Exception thrown: System.Private.CoreLib.dll!System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw
+[ignore]The following are expected but are not being produced in CI builds at this time.
```

Comment:
@davmason, I'm adding unit tests to make sure that our profiler is able to detect unhandled exceptions via the Exception* callback methods. It seems that the `ExceptionUnwindFunctionEnter` is not called once the runtime determines there is an unhandled exception when running in the CI builds or on a local Ubuntu WSL instance. My local Windows machine does have these callbacks though. Any idea what might be causing this behavioral difference?

Reply:
To eliminate the possibility that I'm missing capturing stderr lines at the end of the process, I've set all of the profiler variables manually and run the test app on my Ubuntu WSL instance and can confirm that the unwind callbacks are not called for unhandled exceptions:

```bash
export CORECLR_ENABLE_NOTIFICATION_PROFILERS=1
export CORECLR_ENABLE_PROFILING=1
export CORECLR_PROFILER={6A494330-5848-4A23-9D87-0E57BBF6DE79}
export CORECLR_PROFILER_PATH_64=/home/jander/src/dotnet/dotnet-monitor/artifacts/bin/Linux.x64.Debug/libMonitorProfiler.so
export DotnetMonitorProfiler_RuntimeId=12345
export DotnetMonitorProfiler_StdErrLogger_Level=Trace
dotnet /home/jander/src/dotnet/dotnet-monitor/artifacts/bin/Microsoft.Diagnostics.Monitoring.Profiler.UnitTestApp/Debug/net6.0/Microsoft.Diagnostics.Monitoring.Profiler.UnitTestApp.dll ExceptionThrowCrash

[profiler]dbug: Exception thrown: Microsoft.Diagnostics.Monitoring.Profiler.UnitTestApp.dll!Microsoft.Diagnostics.Monitoring.Profiler.UnitTestApp.Scenarios.ExceptionThrowCrashScenario.Execute
[profiler]dbug: Exception handled: System.CommandLine.dll!System.CommandLine.Invocation.AnonymousCommandHandler+<InvokeAsync>d__6.MoveNext
[profiler]dbug: Exception thrown: System.Private.CoreLib.dll!System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw
[profiler]dbug: Exception handled: System.CommandLine.dll!System.CommandLine.Parsing.ParseResultExtensions+<InvokeAsync>d__0.MoveNext
[profiler]dbug: Exception thrown: System.Private.CoreLib.dll!System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw
[profiler]dbug: Exception handled: System.CommandLine.dll!System.CommandLine.Parsing.ParserExtensions+<InvokeAsync>d__3.MoveNext
[profiler]dbug: Exception thrown: System.Private.CoreLib.dll!System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw
Unhandled exception. System.InvalidOperationException: Operation is not valid due to the current state of the object.
   at Microsoft.Diagnostics.Monitoring.Profiler.UnitTestApp.Scenarios.ExceptionThrowCrashScenario.Execute() in /home/jander/src/dotnet/dotnet-monitor/src/Tests/Microsoft.Diagnostics.Monitoring.Profiler.UnitTestApp/Scenarios/ExceptionThrowCrashScenario.cs:line 24
   at System.CommandLine.Handler.<>c__DisplayClass1_0.<SetHandler>b__0(InvocationContext _)
   at System.CommandLine.Invocation.AnonymousCommandHandler.Invoke(InvocationContext context)
   at System.CommandLine.Invocation.AnonymousCommandHandler.InvokeAsync(InvocationContext context)
   at System.CommandLine.Parsing.ParseResultExtensions.InvokeAsync(ParseResult parseResult, IConsole console, CancellationToken cancellationToken)
   at System.CommandLine.Parsing.ParserExtensions.InvokeAsync(Parser parser, String[] args, IConsole console, CancellationToken cancellationToken)
   at Microsoft.Diagnostics.Monitoring.Profiler.UnitTestApp.Program.<Main>(String[] args)
Aborted
```

Reply:
FYI, this is .NET 6; haven't tried other runtime versions.

This might be a case where unwind behaviors are not always observable for unhandled exceptions (the following is regarding `finally` blocks not executing: https://github.com/dotnet/runtime/issues/67214). If so, this makes this method of unhandled exception detect unreliable.